projects/packages/assets/src/class-script-data.php

```diff
@@ -154,20 +157,20 @@ protected static function get_admin_script_data() {
 	}
 
 	/**
-	 * Get the admin script data.
+	 * Get the public script data.
 	 *
 	 * @return array
 	 */
 	protected static function get_public_script_data() {
 
-		$data = array(
-			'site' => array(
-				'icon'              => self::get_site_icon(),
-				'title'             => self::get_site_title(),
-				'host'              => ( new Host() )->get_known_host_guess(),
-				'is_wpcom_platform' => ( new Host() )->is_wpcom_platform(),
-			),
-		);
+		$data = array();
+
+		// if site site is a P2 site, then add the host information
+		$is_p2_site = str_contains( get_stylesheet(), 'pub/p2' ) || function_exists( '\WPForTeams\is_wpforteams_site' ) && is_wpforteams_site( get_current_blog_id() );  // @phan-suppress-current-line PhanUndeclaredFunction -- We only call the function if it exists.
```

Comment:
If the array is empty by default and we only need the data on p2, maybe this could be moved out of the Jetpack plugin? This would have the advantage of improving performance a bit more, by not even having that `str_contains( get_stylesheet(), 'pub/p2' ) || function_exists( '\WPForTeams\is_wpforteams_site' ) && is_wpforteams_site( get_current_blog_id() );` check on the frontend of every page of every self-hosted site.

Reply:
Good point. Presumably `jetpack-mu-wpcom` is relevant then, from what I can see. Added in caf1923d686fcc643056e94330193a313821e691